segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -585,12 +632,12 @@ public void initialize() throws Exception {
                 txn.markPinned(segmentMetadata);
             }
         }
-        log.debug("SystemJournal[{}] Done applying snapshots.", containerId);
 
         // Validate
         return checkInvariants(systemSnapshot)
                 .thenComposeAsync(v ->
                         validateSystemSnapshotExistsInTxn(txn, systemSnapshot), executor)
+                .thenRun(() -> log.debug("SystemJournal[{}] Done applying snapshots.", containerId))
```

Comment:
Please merge this `thenRun` into the next `thenApplyAsync` to reduce callback overhead.

Reply:
fixed.